src/Voron/Platform/PalDefinitions.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+// ReSharper disable BuiltInTypeReferenceStyle
+
+namespace Voron.Platform
+{
+    public unsafe class PalDefinitions
+    {
+        public static Int32 True => 1;
+        public static Int32 False => 0;
+
+        [SuppressMessage("ReSharper", "UnassignedField.Global")]
+        public struct SystemInformation
+        {
+            public Int32 PageSize;
+            public Int32 CanPrefetch;
```

Comment:
I ended up with `[MarshalAs(UnmanagedType.I4)]` for the bool type to make sure we write int32_t into that bool and not something else (as it might change between OSs and compilers)

Reply:
`I4` is fine, yes

Reply:
I4 doesn't work so I am left with U1

Reply:
The problem is that the moment we use `char` in here, that mess up alignment, padding, etc.
We want to ensure that this is not going to cause us (very subtle) problems down the line, so we want to make the compiler's job as simple as possible.
That means that we have to be strict about alignment, help the compiler avoid the need for padding, etc.

Reply:
so we have to cast from bool to int on the pInvoke calling

Reply:
Or use `int`

Reply:
I originally used Int32.. get back to it ?

Reply:
yes